daphne/src/error/aborts.rs

```diff
@@ -149,22 +149,19 @@ impl DapAbort {
 
     /// Abort due to unexpected value for HTTP content-type header.
     pub fn content_type<S>(req: &DapRequest<S>, expected: DapMediaType) -> Self {
-        let Some(want_content_type) = expected.as_str_for_version(req.version) else {
-            return Self::BadRequest(format!(
-                "unexpected content-type for DAP version {:?}",
-                req.version
-            ));
-        };
+        let want_content_type = expected.as_str_for_version(req.version).unwrap_or_else(|| {
+            unreachable!("unexpected content-type for DAP version {:?}", req.version)
+        });
 
-        let Some(got_content_type) = req
+        let got_content_type = req
             .media_type
             .and_then(|mt| mt.as_str_for_version(req.version))
-        else {
-            return Self::BadRequest(format!(
-                "missing or unexpected content type for DAP version {:?}",
-                req.version
-            ));
-        };
+            .unwrap_or_else(|| {
+                unreachable!(
+                    "missing or unexpected content type for DAP version {:?}",
+                    req.version
+                )
+            });
```

Comment:
If this is `None` then the user requested a media type that this version doesn't support, right? This would mean the previous `BadRequest` code was correct

Reply:
Good catch, done.